tests/unit/test_client.py

```diff
@@ -881,21 +881,6 @@ def __call__(self, *args, **kwargs):
         return http_response
 
 
-def test_trino_result_response_headers():
```

Comment:
this is redundant now since the actual user-facing functionality is that `additional_http_headers` works by merging passed headers to existing headers which is tested in `test_trino_query_response_headers` already.

The use of this test was to ensure that headers were travelling between `TrinoQuery` and `TrinoResult` since older prepared statement code basically passed-through the prepare headers through all layers - which was very simple solution but unclean and couples things together (and also buggy as Trino 398 shows).